transport/stunnel/stunnel.go

```diff
@@ -36,8 +36,12 @@ func getImage(options *transport.Options) string {
 	}
 }
 
-func getResourceName(obj types.NamespacedName, component, suffix string) string {
-	return fmt.Sprintf("%s-%s-%s", obj.Name, component, suffix)
+func getResourceName(obj types.NamespacedName, component, prefix string) string {
+	resourceName := fmt.Sprintf("%s-%s-%s", prefix, component, obj.Name)
+	if len(resourceName) > 62 {
+		return resourceName[:62]
+	}
+	return resourceName
```

Comment:
The above is really just a comment for future thought, not a requirement to fix. :)

Reply:
@JohnStrunk The reason for doing it is that - we inject the ConfigMap, Secret volumes into the Pod definition. The only distinction between their names is the `component` part which is either `config` or `credentials`. If a resource name is too long and we truncate the component/prefix part, collisions happen because the `namespacedName` is same for both configmaps and secrets, it's the prefix and component that's different.